cpp/daal/src/algorithms/pca/transform/pca_transform_dense_default_batch_impl.i

```diff
@@ -82,9 +82,18 @@ services::Status ComputeInvSigmas(NumericTable * pVariances, TArray<algorithmFPT
         PRAGMA_VECTOR_ALWAYS
         for (size_t varianceId = 0; varianceId < numFeatures; ++varianceId)
         {
-            pInvSigmas[varianceId] = pRawVariances[varianceId] ?
-                                         algorithmFPType(1.0) / daal::internal::Math<algorithmFPType, cpu>::sSqrt(pRawVariances[varianceId]) :
-                                         algorithmFPType(0.0);
+            if (pRawVariances[varianceId] > 0)
+            {
+                pInvSigmas[varianceId] = algorithmFPType(1.0) / daal::internal::Math<algorithmFPType, cpu>::sSqrt(pRawVariances[varianceId]);
+            }
+            else if (pRawVariances[varianceId] < 0)
+            {
+                pInvSigmas[varianceId] = algorithmFPType(1.0) / daal::internal::Math<algorithmFPType, cpu>::sSqrt(-pRawVariances[varianceId]);
```

Comment:
Is it mathematically correct to reverse variance?

Reply:
Values from `pRawVariances` are negative in some cases which is not mathematically correct at all. Anyway, MKL solver returns these values.
May be we should raise an exception in these cases? Is it a better option?

Reply:
Probably, better to cast negative values to zeros?